client/src/app/s-degree-plan-editor/s-degree-plan-editor.component.html

```diff
@@ -240,11 +239,26 @@
       </div>
       <!-- Course List -->
       <div class="course-list">
-        <span class="h6 bold neutral-gray-clr mb16">Courses</span>
-        <mat-chip-list *ngFor="let department of courseList" class="mb8">
-          <mat-chip *ngFor="let course of department.courses" class="chip"
-            >{{ course.department }}{{ course.code }}</mat-chip
+        <span class="h6 bold neutral-gray-clr mb16">Courses taken</span>
+        <mat-chip-list *ngIf="courseList" class="mb8">
+          <mat-chip
+            *ngFor="let course of courseList; index as i"
+            [matMenuTriggerFor]="menu"
+            class="chip"
           >
+            {{ course.department }}{{ course.code }}
+            <mat-menu #menu="matMenu" style="padding: 0px;">
+              <app-popup
+                [title]="course.school + ' ' + course.department + course.code"
+                [subtitle]="course.title"
+                contentTitle="Description"
+              >
+                <p *ngIf="course?.description">
+                  {{ course.description }}
+                </p>
+              </app-popup>
+            </mat-menu>
+          </mat-chip>
```

Comment:
In order for the app-popup to use the course data, it has to be wrapped within the mat-chip component where we define what course is in *ngIf="let course of courseList"